src/CommonITILTask.php

```diff
@@ -291,10 +291,45 @@ private function handleTaskDuration(array &$input, int $timestart, int $timeend)
         }
     }
 
+    public function assignTechFromtask($input, string $type)
+    {
+        //if user or group assigned to ticket task, add it to the ticket
+        $itemtype = $this->getItilObjectItemType();
+        $item = new $itemtype();
+        $itemData = [
+            'users_id_tech' => new $item->userlinkclass(),
+            'groups_id_tech' => new $item->grouplinkclass(),
+        ];
+        $foreignkey = getForeignKeyFieldForItemType($itemtype);
+        if (isset($input[$foreignkey])) {
```

Comment:
When it is possible, please try to do something like this:
```php
if (!isset($input[$foreignkey])) {
    return;
}
```

It will remove some indentation levels and make the code easier to read.